beacon-chain/blockchain/process_block.go

```diff
@@ -202,17 +204,72 @@ func (s *Service) onBlockInitialSyncStateTransition(ctx context.Context, signed
 	if err != nil {
 		return errors.Wrap(err, "could not execute state transition")
 	}
+	return s.handlePostStateInSync(ctx, signed, blockRoot, postState)
```

Comment:
Did we need this before?

Reply:
This was to make it easier to break up these large functions , so that I could reuse portions of them when batch verifying them. Also these are currently very long methods which do a lot of things, which makes it hard to follow what is going on in them.